src/java/com/puppetlabs/puppetserver/pool/LockablePool.java

```diff
@@ -0,0 +1,25 @@
+package com.puppetlabs.puppetserver.pool;
+
+import java.util.concurrent.TimeUnit;
+
+public interface LockablePool<E> {
+    public void register(E e) throws InterruptedException;
+
+    public E borrowItem() throws InterruptedException;
+
+    public E borrowItemWithTimeout(long timout, TimeUnit unit) throws InterruptedException;
+
+    public void returnItem(E e) throws InterruptedException;
+
+    public void insertPill(E e) throws InterruptedException;
+
+    public void clear();
+
+    public int remainingCapacity();
+
+    public int size();
```

Comment:
I'm not sure whether `clear`, `remainingCapacity`, and `size` should be in this interface?